extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/ServerSideEncryptingAmazonS3.java

```diff
@@ -119,4 +130,16 @@ public void deleteObject(String bucket, String key)
   {
     amazonS3.deleteObject(bucket, key);
   }
+
+  public boolean upload(PutObjectRequest request) throws java.lang.InterruptedException
+  {
+    if (transferManager == null) {
+      putObject(request);
+    } else {
+      Upload transfer = transferManager.upload(serverSideEncryption.decorate(request));
+      transfer.waitForCompletion();
+    }
+    return true;
```

Comment:
From what I can tell, this method always returns `true`.  If so, it should just be changed to a `void` return type.

Reply:
> From what I can tell, this method always returns `true`. If so, it should just be changed to a `void` return type.

I put void as the return type previously, but I could not get EasyMock to work (:- Googled about ways to do it but in vain. Any help would be appreciated! I tried 
```
   EasyMock.expect(s3Client.upload(EasyMock.anyObject(PutObjectRequest.class)))
      .andVoid()
      .once();
```

```
   EasyMock.expect(s3Client.upload(EasyMock.anyObject(PutObjectRequest.class)))
      .andAnswer(() -> {
         return null;
      })
      .once();
```
Both didn't work

Reply:
How about this:
```java
s3Client.upload(EasyMock.anyObject(PutObjectRequest.class));
EasyMock.expectLastCall().once();
```

See [here](https://www.baeldung.com/easymock-mocking-void-method) for more info.

Reply:
> How about this:
> 
> ```java
> s3Client.upload(EasyMock.anyObject(PutObjectRequest.class));
> EasyMock.expectLastCall().once();
> ```
> 
> See [here](https://www.baeldung.com/easymock-mocking-void-method) for more info.

Thanks Eric, your suggested change worked like a charm!